build.gradle

```diff
@@ -404,6 +408,29 @@ project(':iceberg-mr') {
       exclude group: 'org.apache.calcite.avatica'
     }
   }
+
+  shadowJar {
+    configurations = [project.configurations.compile]
+
+    zip64 true
+
+    // include the LICENSE and NOTICE files for the shaded Jar
+    from(projectDir) {
+      include 'LICENSE'
+      include 'NOTICE'
+    }
+
+    dependencies {
+      include(dependency('com.github.ben-manes.caffeine:caffeine:'))
```

Comment:
why not add dependencies using the `compile` configuration. ? Adding it this way, I think only adds that specific dependency. Also I'm not sure just using a group id is a good idea here as it may include unnecessary jars.

Maybe we can do something similar to what other runtime modules are doing

Reply:
Yeah, as I said in the comment, I've just done this to see what the bare minimum needed to get it working on a cluster is. Once we have it actually working properly in Hive I can fix this up. I can try not specifying any dependencies and inherit all the compile time ones and relocate the problematic ones but my concern is that we're going to then end up having a huge number of clashes with other jars on Hive's classpath which is why I wanted to narrow this down to as few as possible.